grails-app/domain/mama/ng/scheduler/User.groovy

```diff
@@ -0,0 +1,23 @@
+package mama.ng.scheduler
+
+class User {
+
+    String id = UUID.randomUUID().toString()
+    String username
+    String passwordHash
+    String apiKey
+    UserType type = UserType.GENERAL
+    Date dateCreated
+
+    static constraints = {
+        id generator:'assigned'
+        username(blank: false, nullable: false, unique: true)
+        passwordHash(blank: false, nullable: false)
```

Comment:
Do Grails or Groovy have any types or helpers specifically for passwords or authentication? Just very aware that our chance of implementing this correctly ourselves is essentially zero.